core/src/main/java/uk/gov/justice/services/core/extension/AnnotationScanner.java

```diff
@@ -1,44 +1,57 @@
 package uk.gov.justice.services.core.extension;
 
-import uk.gov.justice.services.core.annotation.Component;
+import uk.gov.justice.services.core.annotation.Event;
 import uk.gov.justice.services.core.annotation.ServiceComponent;
 
-import java.util.Optional;
-
 import javax.enterprise.event.Observes;
 import javax.enterprise.inject.Any;
 import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.AnnotatedType;
 import javax.enterprise.inject.spi.Bean;
 import javax.enterprise.inject.spi.BeanManager;
 import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
 import javax.enterprise.util.AnnotationLiteral;
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.gov.justice.services.core.annotation.Component.componentFromServiceComponent;
 
 /**
  * Scans all beans and processes framework specific annotations.
  */
 public class AnnotationScanner implements Extension {
 
+    private List<Object> events = new ArrayList<>();
+
+    @SuppressWarnings("unused")
+    <T> void processAnnotatedType(@Observes final ProcessAnnotatedType<T> pat) {
+        final AnnotatedType<T> annotatedType = pat.getAnnotatedType();
+        if (annotatedType.isAnnotationPresent(Event.class)) {
+            events.add(new EventFoundEvent(annotatedType.getJavaClass(), annotatedType.getAnnotation(Event.class).value()));
+        }
+    }
+
     @SuppressWarnings("unused")
     void afterDeploymentValidation(@Observes final AfterDeploymentValidation event, final BeanManager beanManager) {
         beanManager.getBeans(Object.class, new AnnotationLiteral<Any>() {
             private static final long serialVersionUID = -3118797828842400134L;
-        }).stream().forEach(bean -> getEvent(bean).ifPresent(beanManager::fireEvent));
+        }).stream().forEach(this::processBeansForEvents);
```

Comment:
Could we use the filter and put the If condition of the ServiceComponent annotation class type? ... or isnt the AnnotationLiternal<ServiceComponent> where we can specifiy the type we want?